services/ui-src/src/components/fields/DateField.tsx

```diff
@@ -14,16 +14,20 @@ export const DateField = ({ name, label, sxOverride, ...props }: Props) => {
   const form = useFormContext();
   form.register(name);
 
-  const [displayValue, setDisplayValue] = useState("");
+  const [displayValue, setDisplayValue] = useState<string>("");
+  const [formattedValue, setFormattedValue] = useState<string>("");
 
-  useEffect(() => {
-    if (displayValue) {
-      form.setValue(name, displayValue || "", { shouldValidate: true });
+  const onChangeHandler = (rawValue: string, formattedValue: string) => {
+    setDisplayValue(rawValue);
+    setFormattedValue(formattedValue);
+    const completeDate = checkDateCompleteness(formattedValue);
+    if (completeDate) {
+      form.setValue(name, formattedValue, { shouldValidate: true });
     }
-  }, [displayValue]);
+  };
 
-  const onChangeHandler = async (inputtedString: string) => {
-    setDisplayValue(inputtedString);
+  const onBlurHandler = () => {
+    form.setValue(name, formattedValue, { shouldValidate: true });
   };
```

Comment:
allows us to validate the field on blur so we can tell users that partial dates are not valid.